GitUI/UserControls/RevisionGrid/Columns/AvatarColumnProvider.cs

```diff
@@ -46,12 +47,32 @@ public override void OnCellPainting(DataGridViewCellPaintingEventArgs e, GitRevi
 
             Column.Width = e.CellBounds.Height;
 
-            var padding = DpiUtil.Scale(2);
-            var imageSize = e.CellBounds.Height - padding - padding;
+            var imageSize = e.CellBounds.Height - _padding - _padding;
 
             Image? image;
             var imageTask = _avatarProvider.GetAvatarAsync(revision.AuthorEmail, revision.Author, imageSize);
 
+            Rectangle avatarRect = new(
+                e.CellBounds.Left + _padding,
+                e.CellBounds.Top + _padding,
+                imageSize,
+                imageSize);
+
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Left, avatarRect.Top, 2, 1);
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Left, avatarRect.Top, 1, 2);
+
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Right - 2, avatarRect.Top, 2, 1);
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Right - 1, avatarRect.Top, 1, 2);
+
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Left, avatarRect.Bottom - 1, 2, 1);
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Left, avatarRect.Bottom - 2, 1, 2);
+
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Right - 2, avatarRect.Bottom - 1, 2, 1);
+            e.Graphics.FillRectangle(style.BackBrush, avatarRect.Right - 1, avatarRect.Bottom - 2, 1, 2);
+
```

Comment:
What are these magic numbers? Should those be scaled as well?
Also, if I interpret this code correctly, it would be more efficient to use clipped path here and draw it in a single operation.

Reply:
At first I just moved some code without really understand it.
After some investigation, I have updated with comments so that it is now much clearer and also discovered that moving the code break the goal of this code so I move it back and found another optimization (cache the placeholder image)